proseco/gui.py

```diff
@@ -0,0 +1,358 @@
+    nSigma = opt['Spoiler']['SigErrMultiplier']
+    mag_spoiled = ~ok.copy()
+    mag_check_col = 'mag_spoil_check_{}_{}'.format(nSigma, stype)
+    if mag_check_col not in cone_stars.columns:
+        cone_stars[mag_check_col] = np.zeros_like(not_bad)
+        cone_stars['mag_spoiled_{}_{}'.format(nSigma, stype)] = np.zeros_like(not_bad)
+    mag_spoiled, checked= check_mag_spoilers(cone_stars, ok, opt)
+    cone_stars[mag_check_col] = cone_stars[mag_check_col] | checked
+    cone_stars['mag_spoiled_{}_{}'.format(nSigma, stype)] = (
+        cone_stars['mag_spoiled_{}_{}'.format(nSigma, stype)] | mag_spoiled)
+    bad_pix_dist = check_bad_pixels(cone_stars, ok & ~mag_spoiled, opt)
+    cone_stars['bad_pix_dist_{}'.format(stype)] = bad_pix_dist
+
+    # these star distance checks are in pixels, so just do them for
+    # every roll
+    cone_stars['star_dist_{}_{}'.format(nSigma, stype)] = 9999
+    star_dist = dist_to_bright_spoiler(cone_stars, ok & ~mag_spoiled, nSigma, opt)
+    cone_stars['star_dist_{}_{}'.format(nSigma, stype)] = np.min(
+                [cone_stars['star_dist_{}_{}'.format(nSigma, stype)], star_dist],
+                axis=0)
+
+    if opt['Type'] == 'Acq':
+        minBoxArc = np.ceil(fixedErrorPad(opt['Type'])/5.0)*5
+        minBoxArc = np.max([opt['Select']['MinSearchBox'], minBoxArc])
+        maxBoxArc = np.array(opt['Select']['MaxSearchBox'])
+        maxBoxArc = np.min(maxBoxArc[maxBoxArc >= minBoxArc])
+    else:
+        maxBoxArc = opt['Select']['MaxSearchBox'] # 25
+        minBoxArc = opt['Select']['MinSearchBox']
+
+    starBox = np.min([cone_stars['star_dist_{}_{}'.format(nSigma, stype)],
+                      cone_stars['bad_pix_dist_{}'.format(stype)],
+                      cone_stars['chip_edge_dist_{}'.format(stype)],
+                      cone_stars['fov_edge_dist_{}'.format(stype)] * ARC_2_PIX], axis=0)
+    box_size_arc = ((starBox * PIX_2_ARC) // 5) * 5
+    box_size_arc[box_size_arc > maxBoxArc] = maxBoxArc
+    cone_stars['box_size_arc_{}'.format(stype)] = box_size_arc
+    bad_box = starBox < (minBoxArc * ARC_2_PIX)
+    #cone_stars['bad_box_{}'.format(stype)] = bad_box
+#    if opt['SearchSettings']['DoColumnRegisterCheck']:
+#        badcolumn = check_column(cone_stars, ok & ~mag_spoiled & ~bad_dist, opt, chip_pos)
+#        ok = ok & ~badcolumn
+#    if opt['SearchSettings']['DoBminusVCheck']:
+#        badbv = check_bv(cone_stars, ok & ~mag_spoiled & ~bad_dist)
+#        ok = ok & ~badbv
+    return ok & ~bad_box & ~mag_spoiled
+
+
+def get_mag_errs(cone_stars, opt):
+    caterr = cone_stars['MAG_ACA_ERR'] / 100.
+    caterr = np.min([np.ones(len(caterr))*opt['Inertial']['MaxMagError'], caterr], axis=0)
+    randerr = opt['Inertial']['MagErrRand']
+    magOneSigError = np.sqrt(randerr*randerr + caterr*caterr)
+    return magOneSigError, magOneSigError**2
+
+
+def select_stage_stars(ra, dec, roll, cone_stars):
+
+    stype = 'Guide'
+    opt = STAR_CHAR[stype][0]
+    opt['Type'] = stype
```

Comment:
It seems like `opt['Type']` is hardwired to `Guide` in this module, so can that be factored out entirely for simplicity?

Reply:
Yes.  Indeed.  For the initial import I cut the "Acq" pieces and hardcoded type, but didn't factor out the type logic.